sbe-tool/src/test/c/CComplianceWithVersionTest.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2013-2024 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "code_generation_test_version/car.h"
+#include "code_generation_test_version/messageHeader.h"
+
+int main(void)
+{
```

Comment:
Shouldn't this test access anything from `car.h` or `messageHeader.h`, i.e. what kind of compilation check will it be doing?

Reply:
It's validating that the code in the header files actually compiles.  Without the fix that you merged earlier, this file would fail to compile. We'd see a bunch of this when running cppbuild:

```
/Users/natebradac/src/real-logic/simple-binary-encoding/cppbuild/Release/generated/c/code_generation_test_version/car.h:2705:16: error: expected expression
        return { NULL, 0 };
               ^
```